x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/optimizer/LocalPhysicalPlanOptimizer.java

```diff
@@ -230,7 +230,7 @@ public static boolean canPushToSource(Expression exp) {
 
         private static boolean isAttributePushable(Expression expression, ScalarFunction operation) {
             if (expression instanceof FieldAttribute f && f.getExactInfo().hasExact()) {
-                return true;
+                return f.exactAttribute().field().isAggregatable();
```

Comment:
Question: how does aggregatability relate to being able to push an attribute?

Reply:
field_caps returns two attributes for a field, that are `searchable` and `aggregatable`. 
My understanding is that `searchable` is anything that can be used to perform a search, eg. both a KEYWORD and a TEXT are searchable, even though you can do two different types of search on them; `aggregatable` is things that you can use for aggs, eg. KEYWORD is aggregatable, TEXT is not; since ESQL execution much more similar to aggs than to search, I _guess_ `aggregatable` is what we need, but the question is open for the PR reviewers

Reply:
This is, in essence, about the presence of doc_values: https://www.elastic.co/guide/en/elasticsearch/reference/current/doc-values.html. Searching is about indexed fields, aggregating is about doc_values: two different types of storing a fields' value. One of these characteristics is called "searchable" (storing values in one storage type), the other is "aggregatable" (storing values in a second storage type - doc_values).